### PR TITLE
Fix: Fixed issue with editing properties of multiple items at the same time

### LINKED ISSUE #14900: Bug: Unable to edit the properties of multiple items simultaneously
### Description

When using checkboxes to select multiple items, I am not able to simultaneously change the properties of multiple items.

### Steps To Reproduce

1. Use checkboxes to select multiple items
2. Right click and select properties
3. Select `hidden` in `general > properties`
4. Refresh and none of these files' properties has been changed

### Requirements

The files I selected should be hidden.

### Files Version

3.2.3.0

### Windows Version

10.0.19045.4123

### Log File

[debug.log](https://github.com/files-community/Files/files/14489669/debug.log)

### PR DESCRIPTION
**Pull request by mafra99**
**Resolved / Related Issues**
- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #14900 

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [x] Are there any other steps that were used to validate these changes?

**Test Cases**
 Hidden status is saved and displayed correctly directly in the UI if only one file is selected
- Select a file, open properties, select hidden + save
- The file is now displayed directly in the list as hidden
- Open properties again and check whether hidden status is set
 
Hidden status is saved and displayed correctly directly in the UI when multiple files are selected
- Select multiple files, select hidden + save
- The files are now displayed directly in the list as hidden
- Open properties again and check whether hidden status is set

**Screenshots (optional)**
Add screenshots here.

### CODE REVIEW

**Comment by hishitetsu** [comment]
There is a small glitch in the following step.
1. Select hidden and non-hidden files together.
2. Open the properties window.
3. Press the Save button.
4. The list appears as if the hidden attribute has been removed, but in fact it has not.

**Comment by mafra99** [comment]
> It might be better to put the checkbox in the third state when hidden and non-hidden files are selected together.

Good idea, thank you

**Review by hishitetsu — APPROVED** [review]
LGTM

File: src/Files.App/ViewModels/Properties/Items/LibraryProperties.cs [hunk 1]
```diff
@@ -141,25 +141,31 @@ private void ViewModel_PropertyChanged(object sender, System.ComponentModel.Prop
...
 					{
-						NativeFileOperationsHelper.SetFileAttribute(Library.ItemPath, System.IO.FileAttributes.Hidden);
-					}
-					else
-					{
-						NativeFileOperationsHelper.UnsetFileAttribute(Library.ItemPath, System.IO.FileAttributes.Hidden);
+						if ((bool)ViewModel.IsHidden)
+						{
+							NativeFileOperationsHelper.SetFileAttribute(Library.ItemPath, System.IO.FileAttributes.Hidden);
+						}
```
Comment (yaira2) [inline src/Files.App/ViewModels/Properties/Items/LibraryProperties.cs:164]:
```suggestion
							NativeFileOperationsHelper.SetFileAttribute(Library.ItemPath, System.IO.FileAttributes.Hidden);
```

File: src/Files.App/ViewModels/Properties/Items/LibraryProperties.cs [hunk 2]
```diff
@@ -141,25 +141,31 @@ private void ViewModel_PropertyChanged(object sender, System.ComponentModel.Prop
...
+						else
+						{
+							NativeFileOperationsHelper.UnsetFileAttribute(Library.ItemPath, System.IO.FileAttributes.Hidden);
+						}
```
Comment (yaira2) [inline src/Files.App/ViewModels/Properties/Items/LibraryProperties.cs:168]:
```suggestion
							NativeFileOperationsHelper.UnsetFileAttribute(Library.ItemPath, System.IO.FileAttributes.Hidden);
```

File: src/Files.App/ViewModels/Properties/Items/LibraryProperties.cs [hunk 3]
```diff
@@ -141,25 +141,29 @@ private void ViewModel_PropertyChanged(object sender, System.ComponentModel.Prop
...
-					{
-						NativeFileOperationsHelper.UnsetFileAttribute(Library.ItemPath, System.IO.FileAttributes.ReadOnly);
+						if ((bool)ViewModel.IsReadOnly)
+						{
+							NativeFileOperationsHelper.SetFileAttribute(Library.ItemPath, System.IO.FileAttributes.ReadOnly);
+						}
+						else
+						{
+							NativeFileOperationsHelper.UnsetFileAttribute(Library.ItemPath, System.IO.FileAttributes.ReadOnly);
+						}
```
Comment (yaira2) [inline src/Files.App/ViewModels/Properties/Items/LibraryProperties.cs:153]:
```suggestion
						if ((bool)ViewModel.IsReadOnly)
							NativeFileOperationsHelper.SetFileAttribute(Library.ItemPath, System.IO.FileAttributes.ReadOnly);
						else
							NativeFileOperationsHelper.UnsetFileAttribute(Library.ItemPath, System.IO.FileAttributes.ReadOnly);
```

File: src/Files.App/ViewModels/Properties/Items/FolderProperties.cs [hunk 4]
```diff
@@ -200,15 +200,18 @@ private async void ViewModel_PropertyChanged(object sender, System.ComponentMode
...
-							Item.ItemPath, System.IO.FileAttributes.Hidden);
-					}
-					else
-					{
-						NativeFileOperationsHelper.UnsetFileAttribute(
-							Item.ItemPath, System.IO.FileAttributes.Hidden);
+						if ((bool)ViewModel.IsHidden)
+						{
+							NativeFileOperationsHelper.SetFileAttribute(
+								Item.ItemPath, System.IO.FileAttributes.Hidden);
```
Comment (yaira2) [inline src/Files.App/ViewModels/Properties/Items/FolderProperties.cs:208]:
```suggestion
							NativeFileOperationsHelper.SetFileAttribute(Item.ItemPath, System.IO.FileAttributes.Hidden);
```

File: src/Files.App/ViewModels/Properties/Items/FolderProperties.cs [hunk 5]
```diff
@@ -200,15 +200,17 @@ private async void ViewModel_PropertyChanged(object sender, System.ComponentMode
...
-							Item.ItemPath, System.IO.FileAttributes.Hidden);
+						if ((bool)ViewModel.IsHidden)
+						{
+							NativeFileOperationsHelper.SetFileAttribute(Item.ItemPath, System.IO.FileAttributes.Hidden);
+						}
+						else
+						{
+							NativeFileOperationsHelper.UnsetFileAttribute(
+								Item.ItemPath, System.IO.FileAttributes.Hidden);
+						}
```
Comment (yaira2) [inline src/Files.App/ViewModels/Properties/Items/FolderProperties.cs:213]:
```suggestion
							NativeFileOperationsHelper.SetFileAttribute(Item.ItemPath, System.IO.FileAttributes.Hidden);
						else
							NativeFileOperationsHelper.UnsetFileAttribute(Item.ItemPath, System.IO.FileAttributes.Hidden);
```

File: src/Files.App/ViewModels/Properties/Items/FileProperties.cs [hunk 6]
```diff
@@ -275,37 +275,43 @@ private async void ViewModel_PropertyChanged(object sender, System.ComponentMode
...
+								System.IO.FileAttributes.Hidden
+							);
+						}
+						else
+						{
+							NativeFileOperationsHelper.UnsetFileAttribute(
+								Item.ItemPath,
+								System.IO.FileAttributes.Hidden
+							);
+						}
```
Comment (yaira2) [inline src/Files.App/ViewModels/Properties/Items/FileProperties.cs:314]:
```suggestion
						if ((bool)ViewModel.IsHidden)
							NativeFileOperationsHelper.SetFileAttribute(Item.ItemPath,	System.IO.FileAttributes.Hidden);
						else
							NativeFileOperationsHelper.UnsetFileAttribute(Item.ItemPath, System.IO.FileAttributes.Hidden);
```

File: src/Files.App/ViewModels/Properties/Items/FileProperties.cs [hunk 7]
```diff
@@ -275,37 +275,33 @@ private async void ViewModel_PropertyChanged(object sender, System.ComponentMode
...
+								System.IO.FileAttributes.ReadOnly
+							);
+						}
+						else
+						{
+							NativeFileOperationsHelper.UnsetFileAttribute(
+								Item.ItemPath,
+								System.IO.FileAttributes.ReadOnly
+							);
+						}
```
Comment (yaira2) [inline src/Files.App/ViewModels/Properties/Items/FileProperties.cs:293]:
```suggestion
						if ((bool)ViewModel.IsReadOnly)
							NativeFileOperationsHelper.SetFileAttribute(Item.ItemPath, System.IO.FileAttributes.ReadOnly);
						else
							NativeFileOperationsHelper.UnsetFileAttribute(Item.ItemPath, System.IO.FileAttributes.ReadOnly);
```

**Review by yaira2 — APPROVED** [review]
Thank you

**Pull request merged**